app/src/main/res/layout/topic_practice_subtopic.xml

```diff
@@ -13,39 +13,44 @@
       type="org.oppia.android.app.topic.practice.practiceitemviewmodel.TopicPracticeSubtopicViewModel" />
   </data>
 
-  <androidx.constraintlayout.widget.ConstraintLayout
+  <FrameLayout
```

Comment:
To merge all files correctly you can use `ConstraintLayout` (CL) inside `ConstraintLayout` such that the `outer CL` is `match_parent` and `inner CL` width can be either `0dp` or `360dp`. When its `0dp` it will automatically be treated as `match_parent`.

Reply:
Nice Idea. But doesn't this looks like a patch work. I didn't knew that Android doesn't support mathc parent as parameter in viewbinding.

Reply:
We can use `databinding` too and it might be clear, I am just doubtful about the UI rendering performance. Can you check if [this](https://stackoverflow.com/a/55726466) works without any UI glitch? 

How is it related to `accessibility`?

Reply:
I have tried that databinding solution but no luck. Also the accessiblity issue I am talking about is implemented here also [https://github.com/oppia/oppia-android/pull/2508/files#:~:text=viewModel.index%20%2B%201))%7D%22-,android%3AimportantForAccessibility%3D%22%40%7BviewModel.chapterPlayState%20!%3D%20ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES%20%3F%20View.IMPORTANT_FOR_ACCESSIBILITY_YES%20%3A%20View.IMPORTANT_FOR_ACCESSIBILITY_NO%7D%22,-android%3AtextColor%3D%22%40%7BviewModel.chapterPlayState](importantForAccessiblity). Normally the `importantForAccessiblity` will take yes or no as a parameter. But we know that there enums values corresponds to true and false hence we used that logic here. I was aksing can something similar be done

Reply:
@ARJUPTA We cannot use databinding because for doing that we need some condition based on which we can identify if we want to keep `match_parent` or `360dp` width. And here we don't have any such condition. Therefore using `CL` inside `CL` is the easiest working solution here.

Reply:
I have tried the nested `ConstraintLayout` solution and it is working correctly. Below is the sample code.
Note that you will need to introduce the correct values for `width`, `margin_start` and `margin_end` dimens in all 4 dimens files and also make sure that the names are also correct.


```
<androidx.constraintlayout.widget.ConstraintLayout
    android:layout_width="match_parent"
    android:layout_height="wrap_content">

    <androidx.constraintlayout.widget.ConstraintLayout
      android:layout_width="@dimen/width"
      android:layout_height="wrap_content"
      android:layout_marginStart="@dimen/margin_start"
      android:layout_marginEnd="@dimen/margin_end"
      app:barrierDirection="end"
      app:layout_constraintBottom_toBottomOf="parent"
      app:layout_constraintEnd_toEndOf="parent"
      app:layout_constraintStart_toStartOf="parent"
      app:layout_constraintTop_toTopOf="parent">

      <androidx.constraintlayout.widget.Barrier
```

Reply:
> @ARJUPTA We cannot use databinding because for doing that we need some condition based on which we can identify if we want to keep `match_parent` or `360dp` width. And here we don't have any such condition. Therefore using `CL` inside `CL` is the easiest working solution here.

@rt4914 I agree with your findings and hence @kritigupta45 can go with making a CL under a Cl for this PR. Although I would like to know that, if we assume there is an observable boolean in the viewmodel which changes its value based on the device orientation (true for portrait & false for landscape). So can we use this boolean to change the constraint between match_parent and some other value (defined in dimens.xml). And if the answer is Yes, then please do share a code snippet.

Reply:
Done.

Reply:
@ARJUPTA Yes we can do that but that would make code more complex as compared to what we have done right now.